crates/driver/src/domain/competition/mod.rs

```diff
@@ -46,7 +46,9 @@ pub struct Competition {
 
 impl Competition {
     /// Solve an auction as part of this competition.
-    pub async fn solve(&self, auction: &Auction) -> Result<Solved, Error> {
+    pub async fn solve(&self, auction: Auction) -> Result<Solved, Error> {
+        let auction = auction.prioritize(&self.eth).await;
```

Comment:
Looks like changing the auction after passing it into this function results in a few edge cases.
Could this be done before hand instead?

Reply:
Any more thoughts on this?